homework-custom-lodash/zip.js

```diff
@@ -0,0 +1,49 @@
+const map = require("./map.js");
+const filter = require("./filter.js");
+
+const baseProperty = function(key) {
+  return object => (object == null ? undefined : object[key]);
+};
+
+const isLength = function(value) {
+  const MAX_SAFE_INTEGER = 9007199254740991;
+  return (
+    typeof value == "number" &&
+    value > -1 &&
+    value % 1 == 0 &&
+    value <= MAX_SAFE_INTEGER
+  );
+};
+
+const isObjectLike = function(value) {
+  return typeof value == "object" && value !== null;
+};
+
+const isArrayLike = function(value) {
```

Comment:
so yes, u have really a lot of such cheks, probably better to reuse this code everywhere